situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/processor/oteltrace/model/OTelProtoHelper.java

```diff
@@ -0,0 +1,145 @@
+    static final String INSTRUMENTATION_LIBRARY_VERSION = "instrumentationLibrary.version";
+    static final String STATUS_CODE = "status.code";
+    static final String STATUS_MESSAGE = "status.message";
+    /**
+     * To make it ES friendly we will replace '.' in keys with '@' in all the Keys in {@link io.opentelemetry.proto.common.v1.KeyValue}
+     */
+    private static final String DOT = ".";
+    private static final String AT = "@";
+    public final static Function<String, String> REPLACE_DOT_WITH_AT = i -> i.replace(DOT, AT);
+    /**
+     * Span and Resource attributes are essential for kibana so they should not be nested. SO we will prefix them with "span.attributes"
+     * and "resource.attributes".
+     *
+     */
+    public final static Function<String, String> SPAN_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> SPAN_ATTRIBUTES + DOT + i.replace(DOT, AT);
+    public final static Function<String, String> RESOURCE_ATTRIBUTES_REPLACE_DOT_WITH_AT = i -> RESOURCE_ATTRIBUTES + DOT + i.replace(DOT, AT);
+
+
+    public static Object convertAnyValue(final AnyValue value) {
+        switch (value.getValueCase()) {
+            case VALUE_NOT_SET:
+            case STRING_VALUE:
+                return value.getStringValue();
+            case BOOL_VALUE:
+                return value.getBoolValue();
+            case INT_VALUE:
+                return value.getIntValue();
+            case DOUBLE_VALUE:
+                return value.getDoubleValue();
+            /**
+             * Both {@link AnyValue.ARRAY_VALUE_FIELD_NUMBER} and {@link AnyValue.KVLIST_VALUE_FIELD_NUMBER} are
+             * nested objects. Storing them in flatten structure is not elasticsearch friendly. So they are stored
+             * as Json string.
+             */
+            case ARRAY_VALUE:
+                try {
+                    return OBJECT_MAPPER.writeValueAsString(value.getArrayValue().getValuesList().stream()
+                            .map(OTelProtoHelper::convertAnyValue).collect(Collectors.toList()));
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            case KVLIST_VALUE:
+                try {
+                    return OBJECT_MAPPER.writeValueAsString(value.getKvlistValue().getValuesList().stream()
+                            .collect(Collectors.toMap(i -> REPLACE_DOT_WITH_AT.apply(i.getKey()),
+                                    i ->convertAnyValue(i.getValue()))));
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            default:
+                throw new RuntimeException("Unknown case");
+        }
+    }
+
+    public static Map<String, Object> getSpanAttributes(final Span span) {
+        return span.getAttributesList().stream().collect(Collectors.toMap(i -> SPAN_ATTRIBUTES_REPLACE_DOT_WITH_AT.apply(i.getKey()), i -> convertAnyValue(i.getValue())));
```

Comment:
Not a blocker: although its not the same code but it feels like repetitive code, can we do anything ?

Reply:
I wanted it to be tested explicitly rather testing based on input object instance. Also keeping it simple.